Maven 4.0.0-rc-4 fails dependency resolution if distributionManagement is defined in parent

### Affected version

4.0.0-rc-4

### Bug description

Using the Maven 4.0.0-rc-4 wrapper, create a project with a `pom.xml` containing:

```
<parent>
    <groupId>org.apache</groupId>
    <artifactId>apache</artifactId>
    <version>35</version>
</parent>
```
and
```
<dependencies>
    <dependency>
        <groupId>org.apache.hadoop</groupId>
        <artifactId>hadoop-client</artifactId>
        <version>3.4.1</version>
    </dependency>
</dependencies>
```

Run `./mvnw validate`

*Expected*
The project validates successfully (as it does when using Maven 3.9.11 or 4.0.0-rc-3).

*Actual*
The validation fails with:
 Could not collect dependencies for project org.example:untitled17:jar:1.0-SNAPSHOT
[ERROR] java.lang.IllegalArgumentException: Invalid Version Range Request: org.apache:apache:pom:21 < [central (https://repo.maven.apache.org/maven2, default, releases), apache.snapshots.https (https://repository.apache.org/content/repositories/snapshots, default, releases+snapshots), repository.jboss.org (https://repository.jboss.org/nexus/content/groups/public/, default, releases), ${distMgmtSnapshotsId} (${distMgmtSnapshotsUrl}, default, releases+snapshots)]   

*Reproducer*
Reproducer project attached: [untitled17.zip](https://github.com/user-attachments/files/21573918/untitled17.zip)

--- a/impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelBuilder.java
+++ b/impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelBuilder.java
@@ -510,6 +510,7 @@ public class DefaultModelBuilder implements ModelBuilder {
             Model interpolatedModel = interpolateModel(
                     Model.newBuilder()
                             .pomFile(model.getPomFile())
+                            .properties(model.getProperties())
                             .repositories(model.getRepositories())
                             .build(),
                     request,

--- a/impl/maven-impl/src/test/java/org/apache/maven/impl/model/DefaultModelBuilderTest.java
+++ b/impl/maven-impl/src/test/java/org/apache/maven/impl/model/DefaultModelBuilderTest.java
@@ -92,6 +92,7 @@ class DefaultModelBuilderTest {
         assertEquals(1, repositories.size()); // central
 
         Model model = Model.newBuilder()
+                .properties(Map.of("thirdParentRepo", "https://third.repo"))
                 .repositories(Arrays.asList(
                         Repository.newBuilder()
                                 .id("first")
@@ -100,18 +101,25 @@ class DefaultModelBuilderTest {
                         Repository.newBuilder()
                                 .id("second")
                                 .url("${secondParentRepo}")
+                                .build(),
+                        Repository.newBuilder()
+                                .id("third")
+                                .url("${thirdParentRepo}")
                                 .build()))
                 .build();
+
         state.mergeRepositories(model, false);
 
         // after merge
         repositories = (List<RemoteRepository>) repositoriesField.get(state);
-        assertEquals(3, repositories.size());
+        assertEquals(4, repositories.size());
         assertEquals("first", repositories.get(0).getId());
-        assertEquals("https://some.repo", repositories.get(0).getUrl()); // interpolated
+        assertEquals("https://some.repo", repositories.get(0).getUrl()); // interpolated (user properties)
         assertEquals("second", repositories.get(1).getId());
         assertEquals("${secondParentRepo}", repositories.get(1).getUrl()); // un-interpolated (no source)
-        assertEquals("central", repositories.get(2).getId()); // default
+        assertEquals("third", repositories.get(2).getId());
+        assertEquals("https://third.repo", repositories.get(2).getUrl()); // interpolated (own model properties)
+        assertEquals("central", repositories.get(3).getId()); // default
     }
 
     private Path getPom(String name) {

